### PR TITLE
Cleanup where.exe invocation

### PR DESCRIPTION
**Pull request by pakrym**
And add more logs

### CODE REVIEW

File: src/AspNetCoreModuleV2/CommonLib/CommonLib.vcxproj [hunk 1]
```diff
@@ -197,6 +197,7 @@
     <ClInclude Include="exceptions.h" />
     <ClInclude Include="GlobalVersionUtility.h" />
     <ClInclude Include="fx_ver.h" />
+    <ClInclude Include="HandleWrapper.h" />
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/CommonLib.vcxproj]: Delete the cpp file

File: src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp [hunk 2]
```diff
@@ -440,9 +441,11 @@ HOSTFXR_UTILITY::InvokeWhereToFindDotnet()
     SECURITY_ATTRIBUTES securityAttributes;
 
     CHAR                pzFileContents[READ_BUFFER_SIZE];
-    HANDLE              hStdOutReadPipe = INVALID_HANDLE_VALUE;
-    HANDLE              hStdOutWritePipe = INVALID_HANDLE_VALUE;
-    LPWSTR              pwzDotnetName = NULL;
+    HandleWrapper<>     hStdOutReadPipe;
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp]: Weird that you have to have empty generics here.

File: src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp [hunk 3]
```diff
@@ -584,53 +562,37 @@ HOSTFXR_UTILITY::InvokeWhereToFindDotnet()
...
-    {
-        SysFreeString(pwzDotnetName);
+        if (fIsCurrentProcess64Bit == (dwBinaryType == SCS_64BIT_BINARY))
+        {
+            // The bitness of dotnet matched with the current worker process bitness.
+            return std::make_optional(struDotnetSubstring.QueryStr());
+        }
     }
 
+    Finished:
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp]: I don't like that we still need the goto. Macro that returns a value if the results failed?

**Comment by jkotalik** [comment]
build failures too.

File: src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp (hunk 3 above)
Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp]: Nope, we need a THROW_IF* set of macros but that is a bit too much work. I want this PR out so the bug can be fixed first.

Comment (jkotalik) [reply src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp]: Sure.

File: src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+struct InvalidHandleTraits
+{
+    using HandleType = HANDLE;
+    static const HANDLE DefaultHandle;
+    static void Close(HANDLE handle) { CloseHandle(handle); }
+};
+
+// Workaround for VC++ bug https://developercommunity.visualstudio.com/content/problem/33928/constexpr-failing-on-nullptr-v141-compiler-regress.html
+const HANDLE InvalidHandleTraits::DefaultHandle = INVALID_HANDLE_VALUE;
```
Comment (pan-wang) [inline src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: I understand your motivation for wrapping a handle. However, the performance cost is ignorable.

Comment (jkotalik) [reply src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: Clarify? Is the perf difference noticeable?

Comment (pan-wang) [reply src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: Sorry,  typo.  Updated the comments

**Comment by pakrym** [comment]
Release failure: https://github.com/aspnet/IISIntegration/issues/959

File: src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h (hunk 4 above)
Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: Abstraction is completely free, it generates same code as without abstraction.
See https://godbolt.org/g/y8XzDe
Test.A and B methods.

Comment (pan-wang) [reply src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: I meant the constructor and destructor unless compiler put them inline

Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: They are both inlined.

**Pull request merged**